x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe.analyses;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class Classification implements DataFrameAnalysis {
+
+    public static final ParseField NAME = new ParseField("classification");
+
+    public static final ParseField DEPENDENT_VARIABLE = new ParseField("dependent_variable");
+    public static final ParseField PREDICTION_FIELD_NAME = new ParseField("prediction_field_name");
+    public static final ParseField NUM_TOP_CLASSES = new ParseField("num_top_classes");
+    public static final ParseField TRAINING_PERCENT = new ParseField("training_percent");
+
+    private static final ConstructingObjectParser<Classification, Void> LENIENT_PARSER = createParser(true);
+    private static final ConstructingObjectParser<Classification, Void> STRICT_PARSER = createParser(false);
+
+    private static ConstructingObjectParser<Classification, Void> createParser(boolean lenient) {
+        ConstructingObjectParser<Classification, Void> parser = new ConstructingObjectParser<>(
+            NAME.getPreferredName(),
+            lenient,
+            a -> new Classification(
+                (String) a[0],
+                new BoostedTreeParams((Double) a[1], (Double) a[2], (Double) a[3], (Integer) a[4], (Double) a[5]),
+                (String) a[6],
+                (Integer) a[7],
+                (Double) a[8]));
+        parser.declareString(constructorArg(), DEPENDENT_VARIABLE);
+        BoostedTreeParams.declareFields(parser);
```

Comment:
Clever trick for reusing code.

However, this made me wonder whether those params should be in a nested object. It'd be ugly though, wouldn't it?

Reply:
It's a matter of taste ;)
Parsing code would actually become a bit cleaner as I could just declare the `BoostedTreeParams` field here and it would have its own parser.

However, with nested object:
1. we need to double-check which parameters we want to move there. I just picked the obvious ones but maybe e.g. `dependentVariable` should live there as well?
2. we need to add BWC handling

WDYT?

Reply:
Yeah, I agree we can leave it as is.